deploy/README-multicloud.md

```diff
@@ -0,0 +1,100 @@
+# Configuring Service Assurance Framework for Multiple Clouds
+
+Multiple Openstack clouds can be configured to target a single instance of SAF.
+There are a few steps to get this set up:
+
+1. Plan the AMQP address prefixes to use for each cloud
+1. Deploy metrics and notification SmartGateways for each cloud to listen on the
+  corresponding address prefixes
+1. Configure each cloud to send it's metrics and notifications to SAF on the
+  correct address
+
+## AMQP addresses
+
+By default, OSP nodes are configured to send data to the `collectd/telemetry`
+and `collectd/notify` addresses on the AMQP bus; and SAF is configured to
+listen on those addresses for monitoring data. In order to support multiple
+clouds and have the ability to easily identify which cloud genereated which
+monitoring data, each cloud should be configured to send to a unique address.
+
+It is recommended to prefix a cloud identifier to the second part of the
+address. For example:
+
+* collectd/cloud1-telemetry
+* collectd/cloud1-notify
+* collectd/cloud2-telemetry
+* collectd/cloud2-notify
+* collectd/us-east-1-telemetry
+* collectd/us-west-3-telemetry
+* ...etc
+
+## Deploying SmartGateways
+
+Two SmartGateways (one for metrics, one for events) need to be deployed
+for each cloud, configured to listen on the correct AMQP address. For example:
+
+```yaml
+apiVersion: smartgateway.infra.watch/v1alpha1
+kind: SmartGateway
+metadata:
+  name: cloud1-telemetry
+spec:
+  amqp_url: qdr-white.sa-telemetry.svc.cluster.local:5672/collectd/cloud1-telemetry
+  serviceType: metrics
```

Comment:
So the Ansible Operator will convert the camel case to snake case for you, but I find it weird that our examples here are a mix of snake and camel case.

In the manifests I've just been generating them into snake case so there is no reliance on a translation mechanism.

https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/user-guide.md#define-the-memcached-spec

Reply:
Fixing this, but I must admit that I got a bit confused looking into which is actually used because the CRD shows camelCase: https://github.com/redhat-service-assurance/telemetry-framework/blob/b15c678fc60b3545d95b85739d3a2af5109b35a1/deploy/operators/smartgateway/crds/smartgateway_v1alpha1_smartgateway_crd.yaml#L37  whereas the CRs we actually deploy has snake_case: https://github.com/redhat-service-assurance/telemetry-framework/blob/b15c678fc60b3545d95b85739d3a2af5109b35a1/deploy/operators/smartgateway/crds/smartgateway_v1alpha1_smartgateway_cr.yaml#L7 

To make things more confusing, the snake_case keys from the CR get translated BACK into camelCase for the config: https://github.com/redhat-service-assurance/smart-gateway-operator/blob/1ea1f822d331a96fc62f11045ed2795255884aac/roles/smartgateway/templates/events-configmap.yaml.j2#L13  I think Ansible is the odd one out here, and it's going to be confusing no matter what we do - your recommendation to avoid the auto-translation seems reasonable, but I'm not sure we've been consistent.

As an aside, I'm also noticing that the CRD still also says "AMQPMetricUrl" - so is it just not used when validating a resource going into the cluster?

Reply:
Honestly I want to figure out how to have the CRD generated for us based on the actual code implementation. At this point the CRD is actually quite out of date based on the most recent set of changes. I'm not sure how to handle this at this point. The alternative is to just strip out all the parameters listed because I'm not even sure the validation happens since you'd expect to have started to see errors deploying once we changed from `AMQPMetricsUrl` to `AMQPUrl`.

Reply:
And yea, I agree that the Ansible operator approach is pretty confusing...

I don't have a strong feeling in either case, because I see validity to both approaches: 1) just set the inbound values as what you would expect them to be after inbound and outbound translations, or 2) set the values to what Ansible is going to expect to convert them to anyways so that we don't cause some sort of break, but results in confusion when the outbound translation converts it back to camelCase.